File: 2023/June/Day 10/decodeTheMessage.cpp
/*

 Source : https://leetcode.com/problems/decode-the-message/submissions/



*/

class Solution {
public:
 
    string stripSpace(string generic)
    {
        string out="";
        for(auto&i:generic)
        {
            if(isalpha(i))
                out+=i;
        }
        return out;
    }
    
    map<char,char> generateMap(string key)
    {
        map<char,char> m;
        char start='a';
        
        for(auto&i:key)
        {
            if(!isalpha(m[i]))
            {
                m[i]=start++;
            }
            
        }
        return m;
    }
    
    string decodeMessage(string key, string message) {
        
        key=stripSpace(key);
        map<char,char> m = generateMap(key);
        string out="";
        for(auto&i:message)
        {
            if(isalpha(i))
            {
                out+=m[i];
            }
            else
                out+=i;
        }
        
        return out;
    }
};

File: 2023/July/Day 3/decodeTheMessage.py
class Solution:

    def preProcess(self,key:str) -> str:
        a="";
        for i in key:
            if i==' ':
                continue
            a=a+i
        return a

    def decodeMessage(self, key: str, message: str) -> str:
        dicts={}
        start='a'
        a=self.preProcess(key)
        for i in a:
            if i not in dicts:
                dicts[i]=start;
                start=chr(ord(start) + 1)
        out="";
        for i in message:
            if(i!=' '):
                out=out+dicts[i];
            else:
                out += i
        return out